app.js

```diff
@@ -108,81 +119,168 @@ app.get('/auth/google', passport.authenticate('google', {
+        res.end(JSON.stringify(data));
     }).otherwise(function (err) {
-        res.send("No search results");
-    })
-})
+        res.end("No search results");
+    });
+});
+
+app.get('/otherProfile', isAuthenticated, function (req, res) {
+    res.sendFile(__dirname + "/views/html/" + "otherProfile.html");
+});
+
+app.post('/otherProfile', isAuthenticated, function (req, res) {
+    if (!req.hasOwnProperty('body') || !req.body.hasOwnProperty('othPro'))
+        res.send("Error.Please try again later");
+    else {
+        console.log(req.body.othPro);
+        res.redirect("/ticket");
+    }
+});
+
+app.get('/ticket', isAuthenticated, function (req, res) {
+    res.sendFile(__dirname + "/views/html/" + "ticket.html");
+});
+
+app.post('/ticket', isAuthenticated, function (req, res) {
+    res.writeHead(200, { 'Content-Type': 'application/json' });
+    if (!req.hasOwnProperty('body') || !req.hasOwnProperty('user') || req.user.length <= 0)
+        res.end(JSON.stringify("Error Please try again later"));
+    else {
+        storageModule.createTicket(req.body.citName, req.body.citEmail, req.body.orgName, req.body.orgEmail, req.body.startDate, req.body.endDate, req.body.TDescr, req.body.type).then(function (data) {
+            res.end(JSON.stringify(data));
+        }).otherwise(function (err) {
+            res.end(JSON.stringify("Error Please try again later"));
+        });
+    }
+});
 
-app.get('/editProfile', function (req, res) {
+app.get('/editProfile', isAuthenticated, function (req, res) {
     res.sendFile(__dirname + "/views/html/" + "editProfile.html");
 });
 
-app.post('/editProfile', function (req, res) {
-    if (!req.hasOwnProperty('body') || !req.body.hasOwnProperty('name'))
-        console.log("error getting data");
-    storageModule.updateUser(json.stringify(req.body)).then(function (data) {
-        console.log("Succesfully edited!")
+app.post('/editProfile', isAuthenticated, function (req, res) {
+    if (!req.hasOwnProperty('body') || !req.hasOwnProperty('user') || req.user.length <= 0)
+        res.send("Error.Please try again later");
+    storageModule.updateUser(req.body, req.user[0]).then(function (data) {
+        res.send("Successfully edited");
     }).otherwise(function (err) {
-        console.log("Error updating! Please try again later!")
+        res.send("Error updating in. Please try again later!");
     });
-    res.redirect('/search');
-})
+});
+
+app.get('/getCurrUser', isAuthenticated, function (req, res) {
+    if (!req.hasOwnProperty('user') || req.user.length <= 0)
+        res.send("Error. Please try again later");
+    else {
+        res.writeHead(200, { 'Content-Type': 'application/json' });
+        res.end(JSON.stringify(req.user[0]));
+    }
+});
+
+app.get('/othPro', isAuthenticated, function (req, res) {
+    if (!req.hasOwnProperty('body') || !req.query.hasOwnProperty('othPro'))
+        res.send("Error.Please try again later");
```

Comment:
The message should be changed over here as the it is BAD Request from the user.  Trying again later with the same request would not make a different so it should be
res.status(400).send("Bad request. The request could not be understood by the server due to malformed syntax."). The same change is applicable in other locations too. wherever this is used